bufferserver/src/main/java/com/datatorrent/bufferserver/internal/DataList.java

```diff
@@ -834,6 +854,8 @@ protected void release(boolean wait, boolean writer)
         } else {
           future = null;
         }
+      } else if (storage == null) {
+        resumeSuspendedClients(-1);
```

Comment:
With numberOfInMemBlockPermits set to -1, resumeSuspendedClients will not resume publishers.